cmd/internal/flowmode/cluster_builder.go

```diff
@@ -62,12 +62,20 @@ func buildClusterService(opts clusterOptions) (*cluster.Service, error) {
 	if config.AdvertiseAddress == "" {
 		advertiseAddress := fmt.Sprintf("%s:%d", net.ParseIP("127.0.0.1"), listenPort)
 		if opts.EnableClustering {
-			addr, err := advertise.FirstAddress(opts.AdvertiseInterfaces)
+			advertiseInterfaces := opts.AdvertiseInterfaces
+			if useAllInterfaces(advertiseInterfaces) {
+				advertiseInterfaces = nil
+			}
+			addr, err := advertise.FirstAddress(advertiseInterfaces)
 			if err != nil {
-				level.Warn(opts.Log).Log("msg", "could not find advertise address using default interface names, "+
+				level.Warn(opts.Log).Log("msg", "could not find advertise address using network interfaces", advertiseInterfaces,
```

Comment:
nit: let's log the opts.AdvertiseInterfaces parameter instead. That way, they will see `all` in logs (or whatever custom handling we might add in the future.

```suggestion
				level.Warn(opts.Log).Log("msg", "could not find advertise address using network interfaces", opts.AdvertiseInterfaces,
```